cpp/src/io/parquet/writer_impl.cu

```diff
@@ -1334,6 +1336,10 @@ void encode_pages(hostdevice_2dvector<gpu::EncColumnChunk>& chunks,
                                 d_chunks_in_batch.flat_view().size_bytes(),
                                 cudaMemcpyDefault,
                                 stream.value()));
+
+  if (comp_stats.has_value()) {
+    comp_stats.value() += collect_compression_statistics(comp_in, comp_res, stream);
+  }
```

Comment:
I'm digging this `operator +=()` business. :]
Question: I'm trying to understand the etiquette for operator usage. Would the following be less readable?
```suggestion
  if (comp_stats) {
    *comp_stats += collect_compression_statistics(comp_in, comp_res, stream);
  }
```

Reply:
I prefer `has_value` as it's more explicit/reads like English.
`value()` vs `*`, don't have a strong preference. I think I just like to make it obvious that the object in question is not a pointer.